### PR TITLE
hide private functionality from docs

### PR DESCRIPTION
**Pull request by joelberkeley**
we hide some private functionality from the docs by commenting out docstrings

### CODE REVIEW

File: trieste/utils/pareto.py [hunk 1]
```diff
@@ -47,23 +48,19 @@ def non_dominated(observations: TensorType) -> tuple[TensorType, TensorType]:
     return tf.boolean_mask(observations, dominance == 0), dominance
 
 
+@dataclass(frozen=True)
```
Comment (joelberkeley) [inline trieste/utils/pareto.py:52]: this was just begging to be a dataclass

File: trieste/acquisition/sampler.py [hunk 2]
```diff
@@ -432,30 +432,24 @@ def __init__(
         else:  # if n <= m  then calculate posterior in gram space (an n*n matrix inversion)
             self._theta_posterior = self._prepare_theta_posterior_in_gram_space()
 
-        self._pre_calc = False  # Flag so we only calculate the posterior for the weights once.
```
Comment (joelberkeley) [inline trieste/acquisition/sampler.py]: i noticed nothing is using this. Should it be used somewhere or deleted? @henrymoss

**Review by hstojic — COMMENTED** [review]
looks good to me, I'm not sure about `self._pre_calc` though - I'll let Henry address that and then its good to go

**Comment by joelberkeley** [comment]
> looks good to me, I'm not sure about `self._pre_calc` though - I'll let Henry address that and then its good to go

i've reverted that so we can merge. Merge when ready

**Comment by hstojic** [comment]
ok, merging if checks are ok

**Review by hstojic — APPROVED** [review]
all good

**Pull request merged**